### ISSUE TITLE
Multi-node training speed is slower than single-node 

### ISSUE THREAD

**Issue by AntonioLiuJia** [mlfoundations/open_clip#466]
When I trained CoCa caption in a single node with 8 v100 32G， the samples_per_second_per_gpu is 30/s/gpu, but when I trained the same configuration with two or three nodes(8 v100 32G), the samples_per_second_per_gpu is fall to 10/s/gpu and 5/s/gpu. 
My training script is:    

torchrun --nproc_per_node=8 --rdzv_endpoint=$HOSTE_NODE_ADDR  --nnodes=2 --node_rank=0 -m training.main  --dataset-type "csv"  --train-data $TRAIN_LIST  --val-data $TEST_LIST  --warmup 3000  --batch-size 24 --precision amp --lr 1e-5  --wd 0.1 --epochs 10 --workers 8 --model "coca_ViT-L-14"  --report-to "wandb"  --local-loss  --gather-with-grad

How do I troubleshoot the problem? Thanks

**Comment by gpucce** [comment]
Hi @AntonioLiuJia I tried in the same setting you posted and I do see a slow down, however it is more limited. Comparing 4 nodes and 1 node with the same setting you post, I get a 20% slow down compared to the very worse one you post. 

This makes me think that it could be a connection issue, indeed I think the per_gpu computation is samplas_per_second_per_gpu metric is just computed by diviing per the number of gpu.

To test this, one thing you could try is 2 gpus in one node and 1 gpu in two nodes and see what is the difference in that case maybe?

**Comment by rwightman** [comment]
@AntonioLiuJia what is the interconnect between the nodes for NCCL?

**Comment by AntonioLiuJia** [comment]
> Hi @AntonioLiuJia I tried in the same setting you posted and I do see a slow down, however it is more limited. Comparing 4 nodes and 1 node with the same setting you post, I get a 20% slow down compared to the very worse one you post.
> 
> This makes me think that it could be a connection issue, indeed I think the samplas_per_second_per_gpu metric is just computed by diviing per the number of gpu.
> 
> To test this, one thing you could try is 2 gpus in one node and 1 gpu in two nodes and see what is the difference in that case maybe?

I test 2 gpus in one node(116.775/s/gpu) and 1 gpu in two nodes(36.4602/s/gpu), Could it be a network bandwidth issue?
my cluster only has a 25Gbps interconnect @rwightman

**Comment by rwightman** [comment]
@AntonioLiuJia yes, too slow, especially since that's likely ethernet. May not be the bandwidth specifically, latency is a big factor too.

**Comment by AntonioLiuJia** [comment]
> @AntonioLiuJia yes, too slow, especially since that's likely ethernet. May not be the bandwidth specifically, latency is a big factor too.

ok, thanks for your reply

**Comment by rom1504** [comment]
Are you using local loss
